From fb18c03edb2660d2a97e1797acef3a9f0f9b545a Mon Sep 17 00:00:00 2001
From: David Koltak <david.koltak@linux.intel.com>
Date: Tue, 26 Jun 2018 10:48:40 -0500
Subject: [PATCH 02/18] FogBugz #571233: Remove sprintf in intel-rsu driver

commit  e8f6a0ff4a5b10965aa1d784ba5bf84462f2bc73 from
https://github.com/altera-opensource/linux-socfpga.git

Fixing Klocwork code analysis violations for banned function
sprintf(). Replacing sprintf() with scnprintf(), which is
considered safer.

Signed-off-by: David Koltak <david.koltak@linux.intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/misc/intel-rsu.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/misc/intel-rsu.c b/drivers/misc/intel-rsu.c
index 3070d931d54b..94b85416f882 100644
--- a/drivers/misc/intel-rsu.c
+++ b/drivers/misc/intel-rsu.c
@@ -130,7 +130,7 @@ static ssize_t current_image_show(struct device *dev,
 	if (!priv)
 		return -ENODEV;
 
-	return sprintf(buf, "%ld", priv->status.current_image);
+	return scnprintf(buf, PAGE_SIZE, "%ld", priv->status.current_image);
 }
 
 /* fail_image_show() - DEVICE_ATTR callback to show fail_image status */
@@ -142,7 +142,7 @@ static ssize_t fail_image_show(struct device *dev,
 	if (!priv)
 		return -ENODEV;
 
-	return sprintf(buf, "%ld", priv->status.fail_image);
+	return scnprintf(buf, PAGE_SIZE, "%ld", priv->status.fail_image);
 }
 
 /* version_show() - DEVICE_ATTR callback to show version status */
@@ -154,7 +154,7 @@ static ssize_t version_show(struct device *dev, struct device_attribute *attr,
 	if (!priv)
 		return -ENODEV;
 
-	return sprintf(buf, "%d", priv->status.version);
+	return scnprintf(buf, PAGE_SIZE, "%d", priv->status.version);
 }
 
 /* state_show() - DEVICE_ATTR callback to show state status */
@@ -166,7 +166,7 @@ static ssize_t state_show(struct device *dev, struct device_attribute *attr,
 	if (!priv)
 		return -ENODEV;
 
-	return sprintf(buf, "%d", priv->status.state);
+	return scnprintf(buf, PAGE_SIZE, "%d", priv->status.state);
 }
 
 /* error_location_show() - DEVICE_ATTR callback to show error_location status */
@@ -178,7 +178,7 @@ static ssize_t error_location_show(struct device *dev,
 	if (!priv)
 		return -ENODEV;
 
-	return sprintf(buf, "%d", priv->status.error_location);
+	return scnprintf(buf, PAGE_SIZE, "%d", priv->status.error_location);
 }
 
 /* error_details_show() - DEVICE_ATTR callback to show error_details status */
@@ -190,7 +190,7 @@ static ssize_t error_details_show(struct device *dev,
 	if (!priv)
 		return -ENODEV;
 
-	return sprintf(buf, "%d", priv->status.error_details);
+	return scnprintf(buf, PAGE_SIZE, "%d", priv->status.error_details);
 }
 
 /*
-- 
2.17.1

